tasks/connectors/edgescan/lib/edgescan_vulnerability.rb

```diff
@@ -49,12 +45,33 @@ def to_kenna_finding
           }
         end
 
-        def to_corresponding_kenna_asset
-          { "external_id" => external_asset_id, "tags" => asset.tags }
+        def to_kenna_asset
+          base = {
+            "external_id" => external_asset_id,
+            "tags" => asset["tags"],
+            "application" => asset_application_id
+          }
+          if ip_address?
+            base["ip"] = location
```

Comment:
The IP address has to be provided in a field named `ip_address` rather than `ip` - see the asset section of the KDI v2 spec.: https://help.kennasecurity.com/hc/en-us/articles/360026413111-Kenna-Data-Importer-JSON-Connector-